src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -6496,6 +6503,22 @@ def update_model_version(
                             f"Model version {existing_model_version_in_target_stage.name} has been set to {ModelStages.ARCHIVED.value}."
                         )
 
+            if model_version_update_model.add_tags:
+                self._attach_tags_to_resource(
+                    tag_names=model_version_update_model.add_tags,
+                    resource_id=existing_model_version.id,
+                    resource_type=TaggableResourceTypes.MODEL_VERSION,
+                )
+            model_version_update_model.add_tags = None
+            if model_version_update_model.remove_tags:
+                self._detach_tags_from_resource(
+                    tag_names=model_version_update_model.remove_tags,
+                    resource_id=existing_model_version.id,
+                    resource_type=TaggableResourceTypes.MODEL_VERSION,
+                )
+
+            model_version_update_model.remove_tags = None
+
```

Comment:
Not sure that there is a real explanation behind that. Can you share some Schemas having such "extended" updates? For instance in `run_metadata` we use an extra method to create those, so it is not really closely coupled with the object itself (e.g. `create_run_metadata` is called as one of the steps in `save_artifact` utility).
It might turn out that having those very related entities coupled in one update method is a bit better choice, IMO. But it is definitely not a battle I want to fight in, so if you share examples and reject my explanations - I will rework 👍🏼

Reply:
I think in their first iterations, most schemas had an `update `function which was relatively more extended. However, as I went through the current implementation, I see that this is not the case anymore. I don't think we need to rework anything in the scope of this PR though, ideally, this would be a separate cleaning-up ticket in the future.

Reply:
If we need examples in any case:

- [Code repository schemas](https://github.com/zenml-io/zenml/blob/main/src/zenml/zen_stores/schemas/code_repository_schemas.py): The entire update on the schema happens within the update method not in the SQLZenStore.
- [Stack component schemas](https://github.com/zenml-io/zenml/blob/main/src/zenml/zen_stores/schemas/component_schemas.py): The update on the configuration and labels are happening inside the update function whereas the update on the connectors is happening on the SQLZenStore.,

This is the inconsistency that I find suboptimal. IMO, the update should either happen fully on the `SQLZenStore` or fully on the `update` method of a schema. Otherwise, the code becomes very hard to track and read.